kimchi/src/verifier.rs

```diff
@@ -437,6 +437,36 @@ where
             .chain((0..COLUMNS).map(Column::Witness))
             .chain((0..COLUMNS).map(Column::Coefficient))
             .chain((0..PERMUTS - 1).map(Column::Permutation))
+            .chain(if index.range_check0_comm.is_some() {
+                Some(Column::Index(GateType::RangeCheck0))
+            } else {
+                None
+            })
+            .chain(if index.range_check1_comm.is_some() {
+                Some(Column::Index(GateType::RangeCheck1))
+            } else {
+                None
+            })
+            .chain(if index.foreign_field_add_comm.is_some() {
+                Some(Column::Index(GateType::ForeignFieldAdd))
+            } else {
+                None
+            })
+            .chain(if index.foreign_field_mul_comm.is_some() {
+                Some(Column::Index(GateType::ForeignFieldMul))
+            } else {
+                None
+            })
+            .chain(if index.xor_comm.is_some() {
+                Some(Column::Index(GateType::Xor16))
+            } else {
+                None
+            })
+            .chain(if index.rot_comm.is_some() {
+                Some(Column::Index(GateType::Rot64))
+            } else {
+                None
+            })
```

Comment:
Suggestion: write each optional commitment chain call using `Option.map`
```
            .chain(index.range_check0_comm.map(|_| Column::Index(GateType::RangeCheck0)))          
```

Bonus(?) point: it makes it look like the rest of the code

Reply:
Pushed the changes